src/main/java/org/jabref/gui/fieldeditors/CitationKeyEditor.java

```diff
@@ -20,41 +20,41 @@
 import org.jabref.preferences.PreferencesService;
 
 import com.airhacks.afterburner.views.ViewLoader;
+import jakarta.inject.Inject;
 
 public class CitationKeyEditor extends HBox implements FieldEditorFX {
 
-    private final PreferencesService preferences;
     @FXML private final CitationKeyEditorViewModel viewModel;
     @FXML private Button generateCitationKeyButton;
     @FXML private EditorTextField textField;
 
+    @Inject private PreferencesService preferencesService;
+    @Inject private DialogService dialogService;
+    @Inject private UndoManager undoManager;
+
     public CitationKeyEditor(Field field,
-                             PreferencesService preferences,
                              SuggestionProvider<?> suggestionProvider,
                              FieldCheckers fieldCheckers,
-                             BibDatabaseContext databaseContext,
-                             UndoManager undoManager,
-                             DialogService dialogService) {
+                             BibDatabaseContext databaseContext) {
+
+        ViewLoader.view(this)
+                  .root(this)
+                  .load();
 
-        this.preferences = preferences;
         this.viewModel = new CitationKeyEditorViewModel(
```

Comment:
we really need to make sure that this does not throw an NPE or such thing at runtime

Reply:
I think I catched all the NPEs in the interface.
Happens only if the ViewLoader.load is called after the ViewModel is created, because DI is done by the ViewLoader. Must always be the first thing in the Constructor. Except if you use the FXML to reference stuff directly in the ViewModel. In that case you have to first call `Injector.registerExistingAndInject(this);` to do DI, then load the ViewModel, then call ViewLoader.load to parse the FXML

Reply:
See IdentifierEditor. Took me a bit, until I fixed that one.